<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1113</width>
    <height>636</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>20</y>
      <width>651</width>
      <height>531</height>
     </rect>
    </property>
    <property name="title">
     <string>GroupBox</string>
    </property>
    <widget class="QPushButton" name="btn_cangtochucdoanhmau">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>210</y>
       <width>141</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Căng tổ chức đồ ảnh màu</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_cangtochucdo">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>210</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Căng tổ chức đồ:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_phannguong">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>160</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Mức phân ngưỡng</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_dosang">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>80</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Độ sáng:</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>265</width>
       <height>25</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="lb_file">
          <property name="text">
           <string>File ảnh:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="In_text"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="btn_phannguong">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>160</y>
       <width>101</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Phân ngưỡng</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ambanxam">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>120</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Âm bản trắng đen</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_amban">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>120</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Âm bản</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_chinhsang">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>80</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Chỉnh sáng</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="In_cangtochucdo">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>210</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="In_dosang">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>211</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_ambanmau">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>120</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Âm bản màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_cangtochucdo">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>210</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Căng tổ chức đồ ảnh xám</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="In_pn">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>160</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_vetochucdoxam">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>260</y>
       <width>171</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Vẽ tổ chức đồ cho ảnh mức xám</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_tochucdoanhmau">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>290</y>
       <width>171</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Vẽ tổ chức đồ cho ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_biendoituyentinh">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>260</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Biến đổi tuyến tính tổ chức đồ</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_canbangtochucdo">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>290</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Cân bằng tổ chức đồ</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_kthuocmatna">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>330</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Kích thước mặt nạ:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lb_tilenhieu">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>360</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Tỉ lệ nhiễu:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="In_kichthuoc">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>330</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="In_tile">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>360</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loctrungbinh">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>330</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc trung bình</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_loctrungvi">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>330</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Lọc trung vị</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_themnhieu">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>360</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Thêm nhiễu ảnh xám</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_themnhieuanhmau">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>360</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Thêm nhiễu ảnh màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gradient">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>400</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Gradient</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gradientmau">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>440</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Gradient màu</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_robert">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>400</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Robert</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>440</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>PushButton</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_sobel">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>400</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Sobel</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_prewitt">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>400</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Prewitt</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_laplace">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>400</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Laplace</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_gausslaplace">
     <property name="geometry">
      <rect>
       <x>510</x>
       <y>400</y>
       <width>91</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Gauss + Laplace</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_chonanh">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>30</y>
       <width>111</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Chọn ảnh</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_tuantu">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>150</y>
       <width>131</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Giải thuật tuần tự</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_3">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>180</y>
       <width>131</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Giải thuật Otsu</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1113</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
